src/main/scala/mesosphere/marathon/raml/IpcModeConversion.scala

```diff
@@ -0,0 +1,26 @@
+package mesosphere.marathon
+package raml
+
+/**
+  * Conversion from [[state.IpcMode]] to [[raml.IPCMode]] and vice versa.
+  */
+trait IpcModeConversion {
+
+  implicit val ramlIpcModeRead = Reads[raml.IPCMode, state.IpcMode] {
+    case IPCMode.Private => state.IpcMode.Private
+    case IPCMode.ShareParent => state.IpcMode.ShareParent
+  }
+
+  implicit val ramlIpcModeWrite = Writes[state.IpcMode, raml.IPCMode] {
+    case state.IpcMode.Private => IPCMode.Private
+    case state.IpcMode.ShareParent => IPCMode.ShareParent
+  }
+
+  implicit val protoRamlIpcMode = Writes[Protos.ExtendedContainerInfo.LinuxInfo.IpcInfo.IpcMode, raml.IPCMode] {
+    case Protos.ExtendedContainerInfo.LinuxInfo.IpcInfo.IpcMode.PRIVATE => IPCMode.Private
+    case Protos.ExtendedContainerInfo.LinuxInfo.IpcInfo.IpcMode.SHARE_PARENT => IPCMode.ShareParent
+    case badValue => throw new IllegalStateException(s"unsupported ipcMode $badValue")
```

Comment:
Just food for thought:
This is similar to https://jira.mesosphere.com/browse/COPS-4883 – and what we do in all these conversions. I wonder whether there's a "better" way to expose this issue (and fall back to a _safe_ default?). Ideas welcome.

Reply:
Hmmm, i'd assume the issue would only arise if we were to add a new entry to the enum. Maybe there's a way to treat non-exhaustive pattern matching as an error? That would be much better than having an IllegalStateException there